storage/src/nodestore.rs

```diff
@@ -1686,16 +1687,25 @@ impl<S: ReadableStorage> NodeStore<Committed, S> {
 pub(crate) struct FreeListIterator<'a, S: ReadableStorage> {
     storage: &'a S,
     next_addr: Option<LinearAddress>,
+    parent_ptr: FreeListParentPtr,
 }
 
 impl<'a, S: ReadableStorage> FreeListIterator<'a, S> {
-    pub(crate) const fn new(storage: &'a S, next_addr: Option<LinearAddress>) -> Self {
-        Self { storage, next_addr }
+    pub(crate) const fn new(
+        storage: &'a S,
+        next_addr: Option<LinearAddress>,
+        parent_ptr: FreeListParentPtr,
+    ) -> Self {
+        Self {
+            storage,
+            next_addr,
+            parent_ptr,
```

Comment:
I'm not sure I love having the iterator know what context it is running in. Things in the iterator should only iterate. The caller should have the context and map in the missing bits.